pkg/gator/verify/runner.go

```diff
@@ -358,10 +358,9 @@ func (r *Runner) validateAndReviewAdmissionReviewRequest(ctx context.Context, c
 		}
 	}
 
-	// parse into webhook/admission type
 	req := &admission.Request{AdmissionRequest: *ar.Request}
 	if err := util.SetObjectOnDelete(req); err != nil {
-		return nil, fmt.Errorf("%w: %w", gator.ErrNilOldObject, err)
+		return nil, fmt.Errorf("error validating AdmissionReview for DELETE operation: %w", gator.ErrNilOldObject)
```

Comment:
We should not swallow the original error, just in case SetOldObjectOnDelete() changes.

Since k8s v1.15 is long outside its support window, we're probably okay to change the error text of this line:

https://github.com/open-policy-agent/gatekeeper/blob/780a66a75e44d31e8cc6ff15c3297da3f9ae6c04/pkg/util/request_validation.go#L11

to remove the reference.

The original code also assumes every error will be a nil old object error, which is also risky. IMO the best solution is to keep what you have, but wrap the original error (whose text has been modified).

That does mean that this test line will need updating:

https://github.com/open-policy-agent/gatekeeper/blob/780a66a75e44d31e8cc6ff15c3297da3f9ae6c04/pkg/gator/verify/runner_test.go#L1158

Reply:
> which is also risky.

changed the gator err too fyi